guides/source/testing.md

```diff
@@ -1242,378 +1375,320 @@ assert_response :redirect
-
-### Available Request Types for Functional Tests
-
-If you're familiar with the HTTP protocol, you'll know that `get` is a type of request. There are 6 request types supported in Rails functional tests:
-
-* `get`
-* `post`
-* `patch`
-* `put`
-* `head`
-* `delete`
-
-All of request types have equivalent methods that you can use. In a typical C.R.U.D. application you'll be using `get`, `post`, `put`, and `delete` more often.
-
-NOTE: Functional tests do not verify whether the specified request type is accepted by the action, we're more concerned with the result. Request tests exist for this use case to make your tests more purposeful.
+require "test_helper"
 
-### Testing XHR (Ajax) Requests
+class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
+  driven_by :selenium, using: :headless_chrome
+end
+```
 
-To test Ajax requests, you can specify the `xhr: true` option to `get`, `post`,
-`patch`, `put`, and `delete` methods. For example:
+If you want to use a remote browser, e.g. [Headless Chrome in
+Docker](https://github.com/SeleniumHQ/docker-selenium), you have to add a remote
+`url` and set `browser` as remote through `options`.
 
 ```ruby
-test "ajax request" do
-  article = articles(:one)
-  get article_url(article), xhr: true
+require "test_helper"
 
-  assert_equal "hello world", @response.body
-  assert_equal "text/javascript", @response.media_type
+class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
+  url = ENV.fetch("SELENIUM_REMOTE_URL", nil)
+  options = if url
+    { browser: :remote, url: url }
+  else
+    { browser: :chrome }
+  end
+  driven_by :selenium, using: :headless_chrome, options: options
 end
 ```
 
-### The Three Hashes of the Apocalypse
-
-After a request has been made and processed, you will have 3 Hash objects ready for use:
-
-* `cookies` - Any cookies that are set
-* `flash` - Any objects living in the flash
-* `session` - Any object living in session variables
-
-As is the case with normal Hash objects, you can access the values by referencing the keys by string. You can also reference them by symbol name. For example:
+Now you should get a connection to the remote browser.
 
-```ruby
-flash["gordon"]               # or flash[:gordon]
-session["shmession"]          # or session[:shmession]
-cookies["are_good_for_u"]     # or cookies[:are_good_for_u]
+```bash
+$ SELENIUM_REMOTE_URL=http://localhost:4444/wd/hub bin/rails test:system
 ```
 
-### Instance Variables Available
-
-**After** a request is made, you also have access to three instance variables in your functional tests:
-
-* `@controller` - The controller processing the request
-* `@request` - The request object
-* `@response` - The response object
-
+If your application in test is running remote too, e.g. within a Docker
+container, Capybara needs more input about how to [call remote
```

Comment:
"your application in test is running remote too" reads confusing. Maybe "your application is remote"